src/fhost_fpga.py

```diff
@@ -150,36 +157,63 @@ def __init__(self, host, katcp_port=7147, boffile=None, connect=True, config=Non
             self.min_load_time = None
             self.network_latency_adjust = None
 
+        self._pfb_of_counts = {}
+        self._ct_counts = {}
+        self._qdr_counts = {}
+
     @classmethod
     def from_config_source(cls, hostname, katcp_port, config_source):
         boffile = config_source['bitstream']
         return cls(hostname, katcp_port=katcp_port, boffile=boffile,
                    connect=True, config=config_source)
 
-    def ct_okay(self, sleeptime=1):
+    def ct_okay(self, wait_time=1):
         """
-        Is the corner turner working?
-        :return: True or False,
+        Checks if the corner turner working.
+        :param: wait_time : Float or None : If not None, fetch ct counter,
+            wait this long, and fetch a second value; Else, use last read value
+            from cache. Value is not cached if wait_time is None.
+        :return: bool : True, if corner tuner is okay
+
         """
-        ct_ctrs0 = self.registers.ct_ctrs.read()['data']
-        time.sleep(sleeptime)
-        ct_ctrs1 = self.registers.ct_ctrs.read()['data']
+        # ToDo: Should this be hard-coded?
+        for cnt in range(2):
+            if wait_time:
+                ct_ctrs_err0 = (self.registers.ct_ctrs.read()['data']
+                    ['ct_err_cnt{}'.format(cnt)])
+                ct_ctrs_parerr0 = (self.registers.ct_ctrs.read()['data']
+                    ['ct_parerr_cnt{}'.format(cnt)])
+                time.sleep(wait_time)
+            else:
+                ct_ctrs_err0 = self._ct_counts.get(cnt, 0)
+                ct_ctrs_parerr0 = self._ct_counts.get(cnt, 0)
 
-        err0_diff = ct_ctrs1['ct_err_cnt0'] - ct_ctrs0['ct_err_cnt0']
-        err1_diff = ct_ctrs1['ct_err_cnt1'] - ct_ctrs0['ct_err_cnt1']
-        parerr0_diff = ct_ctrs1['ct_parerr_cnt0'] - ct_ctrs0['ct_parerr_cnt0']
-        parerr1_diff = ct_ctrs1['ct_parerr_cnt1'] - ct_ctrs0['ct_parerr_cnt1']
+            ct_ctrs_err1 = (self.registers.ct_ctrs.read()['data']
```

Comment:
Along with comment above,  here you are reading the register twice just to get different fields, whereas you could get both fields from the same read.